src/examples/mfem_common_plane.cpp

```diff
@@ -0,0 +1,429 @@
+      for (int i{0}; i < par_ref_levels; ++i)
+      {
+        pmesh->UniformRefinement();
+      }
+    }
+  }
+  timer.stop();
+  SLIC_INFO_ROOT(axom::fmt::format(
+    "Time to create parallel mesh: {0:f}ms", timer.elapsedTimeInMilliSec()
+  ));
+  
+  // set up data collection for output
+  auto dc_pv = mfem::ParaViewDataCollection("common_plane_pv", pmesh.get());
+  auto dc_vi = mfem::VisItDataCollection("common_plane_vi", pmesh.get());
+
+  // grid function for higher-order nodes
+  timer.start();
+  auto fe_coll = mfem::H1_FECollection(order, pmesh->SpaceDimension());
+  auto par_fe_space = mfem::ParFiniteElementSpace(
+    pmesh.get(), &fe_coll, pmesh->SpaceDimension());
+  auto coords = mfem::ParGridFunction(&par_fe_space);
+  if (order > 1)
+  {
+    pmesh->SetNodalGridFunction(&coords, false);
+  }
+  else
+  {
+    pmesh->GetNodes(coords);
+  }
+  dc_pv.RegisterField("pos", &coords);
+  dc_vi.RegisterField("pos", &coords);
+
+  // grid function for displacement
+  mfem::ParGridFunction u { &par_fe_space };
+  dc_pv.RegisterField("disp", &u);
+  dc_vi.RegisterField("disp", &u);
+  u = 0.0;
+
+  // grid function for velocity
+  mfem::ParGridFunction v { &par_fe_space };
+  dc_pv.RegisterField("vel", &v);
+  dc_vi.RegisterField("vel", &v);
+  v = 0.0;
+  timer.stop();
+  SLIC_INFO_ROOT(axom::fmt::format(
+    "Time to create grid functions: {0:f}ms", timer.elapsedTimeInMilliSec()
+  ));
+
+  // set initial velocity
+  timer.start();
+  {
+    mfem::Array<int> attrib_marker(pmesh->attributes.Max());
+    attrib_marker = 0;
+    for (auto move_attrib : move_attribs)
+    {
+      attrib_marker[move_attrib-1] = 1;
+    }
+    mfem::Array<int> vdof_marker(par_fe_space.GetVSize());
+    vdof_marker = 0;
+    for (int e{0}; e < pmesh->GetNE(); ++e)
+    {
+      if (attrib_marker[pmesh->GetElement(e)->GetAttribute()-1])
+      {
+        mfem::Array<int> vdofs;
+        par_fe_space.GetElementDofs(e, vdofs);
+        par_fe_space.DofsToVDofs(2, vdofs);
+        for (auto vdof : vdofs)
+        {
+          vdof_marker[vdof] = 1;
+        }
+      }
+    }
+    for (int i{0}; i < vdof_marker.Size(); ++i)
+    {
+      if (vdof_marker[i])
+      {
+        v[i] = -std::abs(initial_v);
+      }
+    }
+  }
+
+  // recover dirichlet bc tdof list
+  mfem::Array<int> ess_vdof_list;
+  {
+    mfem::Array<int> ess_vdof_marker;
+    mfem::Array<int> ess_bdr(pmesh->bdr_attributes.Max());
+    ess_bdr = 0;
+    for (auto fix_attrib : fix_attribs)
+    {
+      ess_bdr[fix_attrib-1] = 1;
+    }
+    par_fe_space.GetEssentialVDofs(ess_bdr, ess_vdof_marker);
+    mfem::FiniteElementSpace::MarkerToList(ess_vdof_marker, ess_vdof_list);
+  }
+  timer.stop();
+  SLIC_INFO_ROOT(axom::fmt::format(
+    "Time to set up boundary conditions: {0:f}ms", timer.elapsedTimeInMilliSec()
+  ));
+
+  // set up mfem elasticity bilinear form
+  timer.start();
+  mfem::ConstantCoefficient rho {100.0};
+  mfem::ConstantCoefficient lambda {100000.0};
```

Comment:
We could tune these parameters to a common material used in other tests, for instance aluminum.

Reply:
I made them changeable parameters in the input.